pkg/declarativeconfig/transform/access_scope_test.go

```diff
@@ -0,0 +1,293 @@
+	assert.Empty(t, scopeProto.GetRules().GetNamespaceLabelSelectors())
+	assert.Empty(t, scopeProto.GetRules().GetIncludedClusters())
+	expectedNamespaces := []*storage.SimpleAccessScope_Rules_Namespace{
+		{
+			ClusterName:   "clusterA",
+			NamespaceName: "NamespaceA",
+		},
+		{
+			ClusterName:   "clusterA",
+			NamespaceName: "NamespaceB",
+		},
+		{
+			ClusterName:   "clusterB",
+			NamespaceName: "NamespaceC",
+		},
+		{
+			ClusterName:   "clusterB",
+			NamespaceName: "NamespaceD",
+		},
+	}
+	assert.ElementsMatch(t, expectedNamespaces, scopeProto.GetRules().GetIncludedNamespaces())
+
+	// 7. Access scope with "everything-and-the-kitchen-sink", i.e. cluster/namespace label selectors, clusters, and
+	//    namespaces
+	scopeConfig = &declarativeconfig.AccessScope{
+		Name:        "test-scope",
+		Description: "test description",
+		Rules: declarativeconfig.Rules{
+			IncludedObjects: []declarativeconfig.IncludedObject{
+				{Cluster: "clusterA", Namespaces: []string{"NamespaceA", "NamespaceB"}},
+				{Cluster: "clusterB", Namespaces: []string{"NamespaceC", "NamespaceD"}},
+				{Cluster: "clusterC"},
+			},
+			ClusterLabelSelectors: []declarativeconfig.LabelSelector{
+				{Requirements: []declarativeconfig.Requirement{{
+					Key:      "a",
+					Operator: declarativeconfig.Operator(storage.LabelSelector_IN),
+					Values:   []string{"a", "b", "c"},
+				}}},
+			},
+			NamespaceLabelSelectors: []declarativeconfig.LabelSelector{
+				{Requirements: []declarativeconfig.Requirement{
+					{
+						Key:      "a",
+						Operator: declarativeconfig.Operator(storage.LabelSelector_IN),
+						Values:   []string{"a", "b", "c"}}}}}},
+	}
+	msgs, err = at.Transform(scopeConfig)
+	assert.NoError(t, err)
+	require.Len(t, msgs, 1)
+	assert.Contains(t, msgs, simpleAccessScopeType)
+	msg = msgs[simpleAccessScopeType]
+	require.Len(t, msg, 1)
+	scopeProto, ok = msg[0].(*storage.SimpleAccessScope)
+	require.True(t, ok)
+	assert.Equal(t, scopeConfig.Name, scopeProto.GetName())
+	assert.Equal(t, scopeConfig.Description, scopeProto.GetDescription())
+	assert.Equal(t, storage.Traits_DECLARATIVE, scopeProto.GetTraits().GetOrigin())
+	compareNamespaceLabelSelectors(t, scopeConfig, scopeProto)
+	compareClusterLabelSelectors(t, scopeConfig, scopeProto)
+	assert.Equal(t, []string{"clusterC"}, scopeProto.GetRules().GetIncludedClusters())
+	expectedNamespaces = []*storage.SimpleAccessScope_Rules_Namespace{
+		{
+			ClusterName:   "clusterA",
+			NamespaceName: "NamespaceA",
+		},
+		{
+			ClusterName:   "clusterA",
+			NamespaceName: "NamespaceB",
+		},
+		{
+			ClusterName:   "clusterB",
+			NamespaceName: "NamespaceC",
+		},
+		{
+			ClusterName:   "clusterB",
+			NamespaceName: "NamespaceD",
+		},
+	}
+	assert.ElementsMatch(t, expectedNamespaces, scopeProto.GetRules().GetIncludedNamespaces())
+}
+
+func compareClusterLabelSelectors(t *testing.T, scopeConfig *declarativeconfig.AccessScope, scopeProto *storage.SimpleAccessScope) {
```

Comment:
nit: `compareClusterLabelSelectors` and `compareNamespaceLabelSelectors` are almost identical and can be refactored to use common function